powerscale/provider/storagepool_tier_datasource.go

```diff
@@ -0,0 +1,278 @@
+func NewStoragepoolTierDataSource() datasource.DataSource {
+	return &StoragepoolTierDataSource{}
+}
+
+// StoragepoolTierDataSource defines the data source implementation.
+type StoragepoolTierDataSource struct {
+	client *client.Client
+}
+
+// Metadata describes the data source arguments.
+func (d StoragepoolTierDataSource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_storagepool_tier"
+}
+
+// Schema describes the data source arguments.
+func (d StoragepoolTierDataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		// This description is used by the documentation generator and the language server.
+		MarkdownDescription: "This datasource is used to query the Storagepool tiers from PowerScale array. " +
+			"The information fetched from this datasource can be used for getting the details or for further processing in resource block. " +
+			"PowerScale Storagepool tiers provide the ability to configure Storagepool tiers on the cluster.",
+		Description: "This datasource is used to query the Storagepool tiers from PowerScale array. " +
+			"The information fetched from this datasource can be used for getting the details or for further processing in resource block. " +
+			"PowerScale Storagepool tiers provide the ability to configure Storagepool tiers on the cluster.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description:         "Id of Storagepool tiers. Readonly. Fixed value of \"storagepool_tiers\"",
+				MarkdownDescription: "Id of Storagepool tiers. Readonly. Fixed value of \"storagepool_tiers\"",
+				Computed:            true,
+			},
+			"storagepool_tiers": schema.ListNestedAttribute{
+				Description:         "List of Storagepool tiers",
+				MarkdownDescription: "List of Storagepool tiers",
+				Computed:            true,
+				NestedObject: schema.NestedAttributeObject{
+					Attributes: map[string]schema.Attribute{
+						"id": schema.StringAttribute{
+							Description:         "Id of storagepool tier.",
+							MarkdownDescription: "Id of storagepool tier.",
+							Computed:            true,
+						},
+						"children": schema.ListAttribute{
+							Description:         "Manage I/O optimization settings.",
+							MarkdownDescription: "Manage I/O optimization settings.",
+							Computed:            true,
+							ElementType:         types.StringType,
+						},
+						"lnns": schema.ListAttribute{
+							Description:         "Apply to files with manually-managed I/O optimization settings.",
+							MarkdownDescription: "Apply to files with manually-managed I/O optimization settings.",
+							Computed:            true,
+							ElementType:         types.Int32Type,
+						},
+						"name": schema.StringAttribute{
+							Description:         "Manage protection settings.",
```

Comment:
Nit: Can you please double-check and confirm that these descriptions are correct?

Reply:
Hi @ABHISHEK-SINHA10 can you pls confirm the descriptions for all the other fields too? 
For example, I see that the description of `children` is `Manage I/O optimization settings`.

Reply:
@rounak-adhikary  Thanks for pointing out, I have checked all the description and now it's updated, please check once again